R/GoogleData.R

```diff
@@ -0,0 +1,170 @@
+#' R6 Class containing specific attributes and methods for Google data
+#'
+#' @description Google data specific information for downloading, cleaning
+#'  and processing covid-19 region data for an example Country.
+#'
+#' @source \url{https://github.com/GoogleCloudPlatform/covid-19-open-data}
+#' @export
+#' @concept dataset
+#' @examples
+#' \dontrun{
+#' region <- GoogleData$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' region$return()
+#' }
+GoogleData <- R6::R6Class("GoogleData",
+  inherit = DataClass,
```

Comment:
do we want to inherit from country class and write a custom filter method so we can take advantage of countryside?

Reply:
Playing around with this the current filter methods work the same as they do for the others. I don't think google needs something specific the others don't.  I.e. you can pass France and Italy as regions to google at level 1 and it will filter correctly, or a level 2 region at level 1. Does it need any more functionality that this?

Something I have found is that because filter uses the level defined in the class it throws an error if you want to filter a level 2 dataset for a level 1 region, as the regions specified must be of the same level as used to create the data. Perhapse we could add an optional argument of the level of the region you are specifying? or have it check all region level columns?

Reply:
Will handle in seperate PR into master and make final changes in #303